### ISSUE TITLE
addnode RPC call should return success/failure indicator

### ISSUE THREAD

**Issue by ajtowns** [bitcoin/bitcoin#20552]
Calling `bitcoin-cli addnode i-am-not-an-address onetry` fails immediately and permanently, but does not give any indication that happened. Having `addnode .. onetry` instead block and return either an error or indicate the peer id of the new connection would probably be better. See also https://github.com/bitcoin/bitcoin/pull/19315#pullrequestreview-528084558

**Comment by ajtowns** [comment]
cc @amitiuttarwar @jnewbery @jonatack

**Comment by amitiuttarwar** [comment]
thanks AJ! 

As mentioned in that conversation thread, I've started a [branch](https://github.com/amitiuttarwar/bitcoin/commits/2020-11-open-conn-improvements) to incorporate. The branch builds on 19315, but the feature doesn't need to. Happy for someone to pick this up if interested, otherwise I will see it through after 19315.

**Comment by jonatack** [comment]
I agree. It would be good for addnode to return a JSON object with either an error description or info.
```
{
  "result": (added/removed/onetry)
  "address":
  "subversion":
  "peer_id":
  "error": optional
}

**Comment by jonatack** [comment]
- the help could use some updating, e.g. `node: The node (see getpeerinfo for nodes)` isn't very clear about what it accepts

- case insensitivity for add/remove/onetry would be a nice touch (if others agree); we do this in other RPCs (sendtoaddress, sendmany, send, fundrawtransaction, walletcreatefundedpsbt, bumpfee) for estimate_mode...I think it's a reasonable convenience for params that require the user to enter a string word to select an option

**Comment by jnewbery** [comment]
> case insensitivity for add/remove/onetry would be a nice touch

I strongly disagree with this. There should be a canonical capitalization for all rpc arguments.

**Comment by naumenkogs** [comment]
Concept ACK.

**Comment by laanwj** [comment]
Not much time for 20.0. Should we change the milestone?

> I strongly disagree with this. There should be a canonical capitalization for all rpc arguments.

I agree with this; having malleability in the protocol adds complication. Most straightforward is to treat enums as fixed byte sequences.

Also, case insensitivity is a complex issue in the general unicode case. Sure, RPC arguments are all ASCII but it's one of those seemingly trivial requirements that can inadvertently drag in a ton of dependencies.

**Comment by willcl-ark** [comment]
Opened #30381 to address this issue.
